From b1acb415e8aa417bb54480e6df03f16bc32436ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B6rn=20Gernert?= <mail@bjoern-gernert.de>
Date: Mon, 22 Nov 2021 20:19:36 +0100
Subject: [PATCH] Ignore albums which contain the keyword 'smart_gallery'

---
 json_structs.go | 1 +
 smugmug.go      | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/json_structs.go b/json_structs.go
index ae320e8..0228030 100644
--- a/json_structs.go
+++ b/json_structs.go
@@ -39,6 +39,7 @@ type albumsResponse struct {
 
 type album struct {
 	URLPath string `json:"UrlPath"`
+	Keywords string `json:"Keywords"`
 	Uris    struct {
 		AlbumImages struct {
 			URI string `json:"Uri"`
diff --git a/smugmug.go b/smugmug.go
index 8b03d1a..6e77bd8 100644
--- a/smugmug.go
+++ b/smugmug.go
@@ -6,6 +6,7 @@ import (
 	"html/template"
 	"os"
 	"path/filepath"
+	"strings"
 
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
@@ -181,6 +182,7 @@ func buildFilenameTemplate(filenameTemplate string) (*template.Template, error)
 //       - if not, download
 func (w *Worker) Run() error {
 	var err error
+	var i int = 0
 	w.cfg.username, err = w.currentUser()
 	if err != nil {
 		return fmt.Errorf("Error checking credentials: %v", err)
@@ -197,6 +199,10 @@ func (w *Worker) Run() error {
 
 	for _, album := range albums {
 		folder := filepath.Join(w.cfg.Destination, album.URLPath)
+		if strings.Contains(album.Keywords, "smart_gallery") {
+			i++
+			continue
+		}
 
 		if err := createFolder(folder); err != nil {
 			log.WithError(err).Errorf("cannot create the destination folder %s", folder)
@@ -217,6 +223,8 @@ func (w *Worker) Run() error {
 		w.saveImages(images, folder)
 	}
 
+	fmt.Printf("Ignored %d smart galleries\n", i)
+
 	if w.errors > 0 {
 		return fmt.Errorf("Completed with %d errors, please check logs", w.errors)
 	}
-- 
2.25.1

